stream_alert/alert_processor/outputs/output_base.py

```diff
@@ -243,12 +250,90 @@ def _kms_decrypt(self, data):
             str: Decrypted json string
         """
         try:
-            client = boto3.client('kms', region_name=self.region)
+            client = boto3.client('kms', region_name=self._region)
             response = client.decrypt(CiphertextBlob=data)
             return response['Plaintext']
         except ClientError as err:
             LOGGER.error('an error occurred during credentials decryption: %s', err.response)
 
+    @staticmethod
+    def get_formatted_output_credentials_name(service_name, descriptor):
+        """Formats the output name for this credential by combining the service
+        and the descriptor.
+
+        Args:
+            service_name (str): Service name on output class (i.e. "pagerduty", "demisto")
+            descriptor (str): Service destination (ie: slack channel, pd integration)
+
+        Returns:
+            str: Formatted credential name (ie: slack_ryandchannel)
+        """
+        cred_name = str(service_name)
+
+        # should descriptor be enforced in all rules?
+        if descriptor:
+            cred_name = '{}/{}'.format(cred_name, descriptor)
+
+        return cred_name
+
+    def get_aws_account_id(self):
```

Comment:
since this doesn't really need to be "computed", you could set `self._account_id` to a public property (`self.account_id`) if you'd like and remove this method. Same with the secrets bucket method. Will reduce amount of tests as well. Alternatively, you could use the `@property` for these to make read-only instance properties out of them